### PR TITLE
GeeksforGeeks Scrapper

### PR DESCRIPTION
**Pull request by Juhibhojani**
## Description

> Added geeksforgeeks in the src/scrape_up folder and pandas in requirements.txt

## Checklist

> Before submitting this pull request, kindly verify that the ensuing checkpoints have been reached.

- [x] Have you adhered to the repository's defined coding convention rules?
- [x] Have you updated the 'documentation.md' file with the method/function documentation?
- [x] Have you sent a message along with the result or response?
- [x] Have you used the try-catch technique?
- [x] Has the method/class been added to the documentation (md file)?

## Screenshots

> Uploading a screenshot illustrating the approach you developed for validation is required.
image

## Additional Notes/Comments

> Any additional remarks or suggestions concerning this pull request are welcome.

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

### CODE REVIEW

File: src/scrape_up/geeksforgeeks/geeksforgeeks.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import pandas as pd
+from bs4 import BeautifulSoup
```
Comment (nikhil25803) [inline src/scrape_up/geeksforgeeks/geeksforgeeks.py:2]: Do not use packages like `pandas`. Simply return the file as `JSON`.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]
Also removed un-necessary packages and dependencies.

**Comment by Juhibhojani** [comment]
Done @nikhil25803

**Review by nikhil25803 — CHANGES_REQUESTED** [review]
The methods are not working on my system, it might not also work on other's.
Please try doing it without selenium, it is possible ig.
Just give it a try

**Comment by Juhibhojani** [comment]
Can you show me the error it's displaying, that can be helpful as I did try with just beautiful soup at first but it wasn't successful.

**Comment by Juhibhojani** [comment]
Please try to run the code now, I have made a few changes. @nikhil25803

**Review by nikhil25803 — DISMISSED** [review]
Getting this error. look once.

image

**Comment by nikhil25803** [comment]
Hey @Juhibhojani 
The problem is not with your codebase, actually it is related to the `webdriver`. I just checked
To solve this issue, one have to
+ Visit the official ChromeDriver website: https://chromedriver.chromium.org/home
+ Download the compatible version of ChromeDriver for your Chrome browser.
+ Extract the downloaded archive to a suitable location on your system.
+ Add the directory containing the "chromedriver" executable to your system's PATH environment variable.

With this I can not merge this PR, cause I do not the users to must install webdriver on their system.

Look for an alternative, if that works try implementing that else close this PR.

**Comment by Juhibhojani** [comment]
Yes, I got that and I did try with the requests library but it isn't possible, I can try to download the web driver in the init method, same as leetcode_scrapper. Do you want me to proceed in the same way?  I will try to search for a way around it but I am not sure it will work.

**Comment by nikhil25803** [comment]
Give it a try, we will see if that works.

**Comment by Juhibhojani** [comment]
Due to my on going exams, i am unable to work on this and so this might take a day more.

**Comment by nikhil25803** [comment]
No problem @Juhibhojani |

**Comment by Juhibhojani** [comment]
Hey @nikhil25803 , I have installed ChromeDriverManager in the code itself. Check it out

**Review by nikhil25803 — COMMENTED** [review]
Code is still not working @Juhibhojani

**Comment by Juhibhojani** [comment]
@nikhil25803  I have added another argument in options to configure Webdriver. Can you plesase check it out?

**Comment by nikhil25803** [comment]
Hey @Juhibhojani | Reviewed the changes you made, it will work on a local machine but not as a package. Also it will create problems while integrating it with the AWS Lambda function.

So can't really merge the PR because of the state it is in.

**Comment by Juhibhojani** [comment]
Okay, gotcha. So I guess there is nothing I can do here?

**Comment by Juhibhojani** [comment]
I would love to contribute to this project, I guess will select another webpage that can be scraped without the use of selenium.

**Comment by nikhil25803** [comment]
No @Juhibhojani | We already do have a few modules facing the same issue. Add this as a personal project on your GitHub, you must have learned a lot as well throughout this.

We have a few modules which use `bs4` and `beautifulsoup` like Amazon, hashcode, and many others. You should check them out and see if you can add a method to them. Just raise an issue and tag me, and I'll assign you that.

Closing this PR and the corresponding issue for now.

**Comment by Juhibhojani** [comment]
Okay, Thanks, it was nice working on it.

**Pull request closed without merging**